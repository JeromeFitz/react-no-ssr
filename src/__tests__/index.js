import React from 'react'

import NoSSR from '../'

describe('NoSSR', () => {
  describe('Server', () => {
    describe('onSSR: false', () => {
      const MyComp = () => <div>Hello</div>
      const wrapper = render(
        <NoSSR>
          <MyComp />
        </NoSSR>
      )
      it('Should not show children', () => {
        // expect(wrapper.html()).not.toMatch(/Hello/);
        expect(wrapper).toMatchSnapshot();
      })
    })
    describe('onSSR: true', () => {
      const MyComp = () => <div>Hello</div>
      const Loading = () => <div>Loading...</div>
      const wrapper = render(
        <NoSSR onSSR={<Loading />}>
          <MyComp />
        </NoSSR>
      )
      it('Should show children (Loading)', () => {
        // expect(wrapper.html()).toMatch(/Loading/)
        // expect(wrapper.html()).not.toMatch(/Hello/)
        expect(wrapper).toMatchSnapshot();
      })
    })
  })
  describe('Client', () => {
    describe('onSSR: false; Mounted', () => {
      const MyComp = () => <div>Hello</div>
      const wrapper = mount(
        <NoSSR>
          <MyComp />
        </NoSSR>
      )
      it('Should show children (Hello)', () => {
        // expect(wrapper.html()).toMatch(/Hello/)
        expect(wrapper).toMatchSnapshot();
      })
    })
    describe('onSSR: true; Mounted', () => {
      const MyComp = () => <div>Hello</div>
      const Loading = () => <div>Loading...</div>
      const wrapper = mount(
        <NoSSR onSSR={<Loading />}>
          <MyComp />
        </NoSSR>
      )
      it('Should show children (Hello)', () => {
        // expect(wrapper.html()).not.toMatch(/Loading/)
        // expect(wrapper.html()).toMatch(/Hello/)
        expect(wrapper).toMatchSnapshot();
      })
    })
  })
})
