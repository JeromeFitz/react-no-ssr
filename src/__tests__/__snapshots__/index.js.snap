// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSSR Client onSSR: false; Mounted Should show children (Hello) 1`] = `
<NoSSR>
  <MyComp>
    <div>
      Hello
    </div>
  </MyComp>
</NoSSR>
`;

exports[`NoSSR Client onSSR: true; Mounted Should show children (Hello) 1`] = `
<NoSSR
  onSSR={<Loading />}
>
  <MyComp>
    <div>
      Hello
    </div>
  </MyComp>
</NoSSR>
`;

exports[`NoSSR Server onSSR: false Should not show children 1`] = `<span />`;

exports[`NoSSR Server onSSR: true Should show children (Loading) 1`] = `
<div>
  Loading...
</div>
`;
